### PR TITLE
vite also needs to resolve index.{gjs,gts}

### PR DESCRIPTION
**Pull request by patricklx**
blocked by #1876

### CODE REVIEW

**Review by mansona — CHANGES_REQUESTED** [review]
This will likely be fixed by https://github.com/embroider-build/embroider/pull/1794, or at the very least we should add a failing test somewhere after that is merged before we make any more changes like this 👍

File: tests/vite-app/app/components/fancy.gts [hunk 1]
```diff
@@ -1,2 +1,8 @@
-<template>Yay for gts!</template>
+import WithIndex from './with-index';
```
Comment (patricklx) [inline tests/vite-app/app/components/fancy.gts:1]: this is the import that would fail to resolve

**Comment by patricklx** [comment]
one test failed while fetching node...

**Comment by ef4** [comment]
Idea here is that we should try switching at least the gjs/gts support to rely on vote's own resolve-extensions, so that it composes with vite's built-in index search.

We may or may not be able to do that for hbs given that within app code the hbs support is more complex, given the existence of template-only components, etc.

**Comment by patricklx** [comment]
somehow after adding the extensions to resolve.
the module-resolver resolveComponent is receiving `type: ignore` results...

**Comment by patricklx** [comment]
the last commit makes esbuild resolver behave the same as vite, when we have a `not-found` we report it as external.
now another issue:
all imports that are resolved via `resolveWithinMovedPackage` do not get dep optimization, because the importer is in node_modules. this causes the vite app to use duplicate modules @ef4

**Comment by patricklx** [comment]
2 issues:
* resolveWithinMovedPackage changes importer to be in node-modules, which makes vite/esbuild ignore it.
1. solution: generate links to rewritten-packages that are resolvable via node resolution and just rewrite the request specifier
2. Change the specifier to a special prefix that will be treated specially? How?

* virtual modules start with a prefix, which changes all its imports to /fs/... full path. Can be fixed by using a suffix instead.

**Comment by ef4** [comment]
I think this is going to need pairing at office hours to sort through. Changing from prefix to suffix seems like a workaround for what is actually breaking.

**Comment by patricklx** [comment]
Screenshot 2024-04-04 at 16 00 51

**Comment by patricklx** [comment]
will be fixed by #2029

**Pull request closed without merging**